skopt/space.py

```diff
@@ -17,10 +17,10 @@ def fit(self, X):
         return self
 
     def transform(self, X):
-        return X
+        return np.asarray(X, dtype=np.float)
 
     def inverse_transform(self, Xt):
-        return Xt
+        return np.asarray(Xt, dtype=np.float)
```

Comment:
Not sure we should inverse transform everything to float by default. Shouldn't we instead do that on a case by case basis, in the `Dimension` sub-classes?